pkg/common/log/log.go

```diff
@@ -0,0 +1,177 @@
+package log
+
+import (
+	"strings"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var (
+	// Log is a singleton base logger that can be used across the system,
+	// either directly or to create other loggers with name, with values,
+	// and/or locked to a given log level.
+	// It is initialized to the promise delegation log provided by
+	// sigs.k8s.io/controller-runtime, which points to a no-op (null) logger
+	// until `SetLogger` is called.
+	// This is also useful for mocking the default logger tests.
+	Log logr.Logger = ctrl.Log
+
+	// Level holds the minimum level of messages that should be logged.
+	// It is useful for checking the log level in the code and avoid expensive
+	// tasks only required for some specific levels.
+	// It is setup by the `SetLogger` function.
+	Level LogLevel
```

Comment:
Not sure this is even needed. 

As you know, the `logr.Logger` interface, has this `Enabled()` method. It is not useful to get the actual log level option of a logger instance, but can tell if a logger instance is being enabled or not. 

```go
myLogger = log.Log
fmt.Println(myLogger.Enabled()) // true when log level set to debug or info

myDebugLogger = myLogger.V(1)
fmt.Println(myDebugLogger.Enabled()) // true when log level set to debug 
```
we are using in the kuadrant controller:
https://github.com/Kuadrant/kuadrant-controller/blob/145657005a3a64676281a3799e118c24d47bea83/controllers/apiproduct_controller.go#L70

Reply:
You are totally right. Good catch!

Added https://github.com/Kuadrant/authorino/commit/4f0fe1ce564e2c2dfd648713efe73747a2435835.